### DISCUSSION TITLE
How to change SetOption value as default in user_config_override.h

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#21919] [Q&A]
I want to define/change SetOption default value for several CPU version firmware.bin´s in user_config_override.h
Is ist possible & how to do it ?
Or do I have to alway´s change in Source code ?

And also for Sensor default value.

**Comment by barbudor** [comment]
For most setoption it is possible but you have to find out which define allows that (as you don't list the setoption you are interested in, I can't help you more)

For others, either make a PR to make them override le or use the [`USER_BACKLOG`](https://tasmota.github.io/docs/Compile-your-build/#user_backlog) macro

I don't understand "also for Sensor default value"
There's no default values for SENSOR

**Reply by sfromis** [reply]
In `my_user_config.h` you can find the available `#define` symbols for a lot of the SetOption values.

**Reply by HofeBY** [reply]
@barbudor
currently I want to change settings for device "Reset Configuration" proof

SO142=1 changed in tasmota(x).bin (for WLAN stability Fritz!Box´er)

"#define APP_TIMEZONE 99" "Timezone 99"  ; I know my_looking "user_config.h"
& config in "user_config_override.h" is done,
but it isn´t in Firmware and has to be set after first configuration on every device !

At ESP32 devices I want to have CPU Temperature SO146=1 active as default in Tasmota32(x).bin

On Firmware for Shutter devices I want.
ShutterMode 1
Interlock 1,2
Interlock 1
ShutterRelay1 1
ShutterOpenDuration1
ShutterCloseDuration1
WebButton3 Licht

@sfromis 
but not all

Greetings
HofeBY

**Reply by sfromis** [reply]
Yes, not all, but barbudor already told you how you can handle the others.

Note that when you change the **defaults** with a custom build, this will not change the configured values when OTA flashing. Tasmota will only use the baked-in defaults after a full reset.

When it comes to that "ESP32 temperature", that is a deeply flawed number, and you can't expect it to be useful. I've even seen it negative on one chip, and much higher on same type, a few cm away in a room. Both was right after cool-off period and boot. There's a reason why the default was changed to be off, instead of users getting confused by this crap number. Having a temperature sensor is **not** among the features Espressif claim for the chip, they removed that when realizing how bad it is.

**Reply by HofeBY** [reply]
Yes, but after "Reset Configuration" I want to have defined values, changeble If not wished.

ESP32 CPU temperature is not a calibrated one, but if I messure more than 200°C,
I would expect something is burning and I better control what is burning !
And if I messure CPU temperature below -1°C I better control Waterpipes.

I think that´s usefull.

**Reply by sfromis** [reply]
The problem with the "ESP32 Temperature" is not merely lack of calibration, but that is just does not work reliably. I've even seen mentions of "unexplained large jumps", and when showing the temperature on a local screen (as a feature test), I've right now watched the "temperature" jump up and down by 10 degrees. You are being way to charitable about something the chip does not feature.
